File: Javier_Programacion/ProyectoSpringFInal/src/main/java/com/daw2/proyectospringfinal/model/entity/Usuario.java
package com.daw2.proyectospringfinal.model.entity;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name="usuarios")
public class Usuario {
    private Integer id;
    private String username;
    private String password;
    private String email;
    private String nombre;
    private String apellido1;
    private String apellido2;
    private boolean activo;
    private List<Rol> roles;


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Column(length=15, nullable=false, unique=true)
    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    @Column(length=60, nullable=false)
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Column(length=75, unique=true)
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Column(length=20, nullable=false)
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    @Column(length=20, nullable=false)
    public String getApellido1() {
        return apellido1;
    }

    public void setApellido1(String apellido1) {
        this.apellido1 = apellido1;
    }

    @Column(length=20)
    public String getApellido2() {
        return apellido2;
    }

    public void setApellido2(String apellido2) {
        this.apellido2 = apellido2;
    }

    @Column(nullable=false)
    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }



    @ManyToMany
    @JoinTable(name="usuarios_roles",
            joinColumns= @JoinColumn(name="id_usuario"),
            inverseJoinColumns=@JoinColumn(name="id_rol"),
            uniqueConstraints= {@UniqueConstraint
                    (columnNames= {"id_usuario", "id_rol"})})
    public List<Rol> getRoles() {
        return roles;
    }

    public void setRoles(List<Rol> roles) {
        this.roles = roles;
    }



}

File: ERPDAW2-Juanjo/pojos/Usuario.php
<?php 

class Usuario{

	private $id;
	private $idUsuario;
	private $idRol;
	private $login;
	private $password;
	private $activo;


	
	public function __construct($id, $idUsuario, $idRol, $login, $password, $activo)
	{
		$this->id = $id;
		$this->idUsuario = $idUsuario;
		$this->idRol = $idRol;
		$this->login = $login;
		$this->password = $password;
		$this->activo = $activo;
	}


    
    public function getId()
    {
        return $this->id;
    }

    
    public function setId($id)
    {
        $this->id = $id;

        return $this;
    }

    
    public function getIdUsuario()
    {
        return $this->idUsuario;
    }

    
    public function setIdUsuario($idUsuario)
    {
        $this->idUsuario = $idUsuario;

        return $this;
    }

    
    public function getIdRol()
    {
        return $this->idRol;
    }

    
    public function setIdRol($idRol)
    {
        $this->idRol = $idRol;

        return $this;
    }

    
    public function getLogin()
    {
        return $this->login;
    }

    
    public function setLogin($login)
    {
        $this->login = $login;

        return $this;
    }

    
    public function getPassword()
    {
        return $this->password;
    }

    
    public function setPassword($password)
    {
        $this->password = $password;

        return $this;
    }

    
    public function getActivo()
    {
        return $this->activo;
    }

    
    public function setActivo($activo)
    {
        $this->activo = $activo;

        return $this;
    }
}
 ?>
